libsolidity/codegen/ABIFunctions.cpp

```diff
@@ -1434,6 +1482,23 @@ string ABIFunctions::shiftRightFunction(size_t _numBits, bool _signed)
 	});
 }
 
+std::string ABIFunctions::scaleFixedFunction(int fromFractionalDigits, int toFractionalDigits, bool _signed)
```

Comment:
Should these parameters be signed?

Reply:
No, there cannot be negative fractional digits.

Reply:
I think it probably makes sense changing that, but would do that in a separate PR, because otherwise it may be too many conflicting changes again.

Reply:
Oh, it might mess up the subtraction down below, but it shouldn't really matter since the range is fine either way.